docs/hugo/content/design/ADR-2022-09-Reading-Status-Properties-Of-Resources.md

```diff
@@ -0,0 +1,261 @@
+3. Risk for privilege escalation, where a user in the namespace that doesn't have permission to read resources of type `X` instead asks ASO to read that resource. Azure then reflects that value back to them in the error message.
+4. Uncertain general reconcile behavior. This reference constitutes a dependency on the referenced resource. Does this mean that ideally, the resource holding the reference will be `Watch`-ing the referenced resource and immediately react to changes in its dependencies? Implementing that dynamic watch behavior given the current structure of controller-runtime would be very difficult.
+
+### Option 3: Support reading/writing to a ConfigMap or Secret, and indirect through that
+
+The idea here is that `UserManagedIdentity` would allow exporting a Secret or ConfigMap value, and then the `RoleAssignment` resource would support (optionally) sourcing from that store instead of having the value hardcoded.
+The main difference between this and what we have now is for references we can't replace the entire reference with a `SecretReference`, we need a new type like `OptionalSecretReference`:
+
+We would only opt a select few resource properties into this `OptionalSecretReference` structure. We could (via serialization magic) support new properties adopting this shape in a nonbreaking way, in case there are properties we didn't think
+needed this behavior initially but later we decide they do.
+
+```go
+type OptionalSecretReference struct {
+    Ref *SecretReference `json:"ref,omitempty"`
+
+    Value *string `json:"value,omitempty"`
+}
+```
+
+Shown here using the existing `operatorSpec` + `secrets` approach, but we could support a `config` (or `exportedConfig`?):
+```yaml
+apiVersion: managedidentity.azure.com/v1beta20181130
+kind: UserAssignedIdentity
+metadata:
+  name: sample-uai
+spec:
+  location: Germany West Central
+  owner:
+    name: dev-sample-rg
+  operatorSpec:
+    secrets:
+      principalId:
+        name: uai-secret
+        key: principalid
+---
+apiVersion: authorization.azure.com/v1beta20200801preview
+kind: RoleAssignment
+metadata:
+  name: 6a2d44f5-57d8-4916-9f46-ff7c9c1b338f
+spec:
+  location: Germany West Central
+  owner:
+    name: samplevnet
+    group: network.azure.com
+    kind: VirtualNetwork
+  principalId:
+    ref:
+      name: uai-secret
+      key: principalid
+  roleDefinitionReference:
+    armId: /subscriptions/00000000-0000-0000-0000-000000000000/providers/Microsoft.Authorization/roleDefinitions/b24988ac-6180-42a0-ab88-20f7382dd24c
+```
+
+Pros:
+1. Generic
+2. Would solve another related problem as well (see [2349](https://github.com/Azure/azure-service-operator/issues/2349) and [2350](https://github.com/Azure/azure-service-operator/issues/2350), that of how a customer pod can access some of these variables.
+3. Solves the priviledge escalation risk of option 2
```

Comment:
Break this out as a separate `Security` section, and cover that it resolves the privilege escalation risk by requiring the source resource (the identity) to opt-in to sharing the property.

Reply:
Done